tests/integration/test_fim/test_synchronization/test_sync_overlap.py

```diff
@@ -0,0 +1,193 @@
+        - r".*Previous sync was successful. Sync interval is reset to: '(\\d+)s'"
+
+    tags:
+        - scheduled
+    '''
+
+    wazuh_log_monitor = FileMonitor(LOG_FILE_PATH)
+
+    # If config is invalid, check that invalid config value message appers
+    if metadata['response_timeout'] == 'invalid' or metadata['max_interval'] == 'invalid':
+        wazuh_log_monitor.start(timeout=global_parameters.default_timeout,
+                                callback=generate_monitoring_callback(fim.CB_INVALID_CONFIG_VALUE),
+                                error_message=fim.ERR_MSG_INVALID_CONFIG_VALUE,
+                                update_position=True).result()
+
+    # Wait for new sync
+    wazuh_log_monitor.start(timeout=global_parameters.default_timeout, callback=callback_detect_synchronization,
+                            error_message=fim.ERR_MSG_FIM_SYNC_NOT_DETECTED, update_position=True).result()
+
+    wazuh_log_monitor = FileMonitor(LOG_FILE_PATH)
+
+    # Check if response_timeout has elapsed, and sync is still running, the sync interval is doubled
+    interval = wazuh_log_monitor.start(timeout=global_parameters.default_timeout*5,
+                                       callback=generate_monitoring_callback(fim.CB_SYNC_SKIPPED),
+                                       accum_results=metadata['doubled_times'],
+                                       error_message=fim.ERR_MSG_SYNC_SKIPPED_EVENT, update_position=True).result()
+
+    if metadata['doubled_times'] > 1:
+        new_interval = interval[-1]
+    else:
+        new_interval = interval
+
+    # Check interval when doubled is not higher than max interval, if max_interval is higher than configured interval
+    # Check interval when doubled is equal than configured interval, if max_interval is lower than configured interval
+    if metadata['max_interval'] != 'invalid':
+        if not metadata['lower']:
+            assert int(new_interval) <= int(metadata['max_interval']), f"Invalid value for interval: {new_interval},\
+                                                                         cannot be more than MAX_INTERVAL:\
+                                                                         {metadata['max_interval']}"
+        else:
+            assert int(new_interval) <= int(metadata['interval']), f"Invalid value for interval: {new_interval}, cannot\
+                                                                     be more than interval: {metadata['interval']}"
+
+    # Check when sync ends sync_interval is returned to normal after response_timeout since last message.
+    if not metadata['lower']:
+        wazuh_log_monitor.start(timeout=global_parameters.default_timeout*10,
+                                callback=generate_monitoring_callback(fim.CB_SYNC_INTERVAL_RESET),
+                                error_message=fim.ERR_MSG_SYNC_INTERVAL_RESET_EVENT,
+                                update_position=True).result()
```

Comment:
Also, we can check the config is restored with the initial value too.

Reply:
Fixed in https://github.com/wazuh/wazuh-qa/pull/3380/commits/6b66bc842d17c0bc1f7d20dc3dcc8c96bce3d18e